pkg/virt-launcher/notify-client/client.go

```diff
@@ -248,6 +248,28 @@ func eventCallback(c cli.Connection, domain *api.Domain, libvirtEvent libvirtEve
 		watchEvent := watch.Event{Type: watch.Modified, Object: domain}
 		client.SendDomainEvent(watchEvent)
 		updateEvents(watchEvent, domain, events)
+	case api.ReasonPausedIOError:
+		domainDisksWithErrors, err := d.GetDiskErrors(0)
+		if err != nil {
+			log.Log.Reason(err).Error("Could not get disks with errors")
+		}
+		for _, disk := range domainDisksWithErrors {
+			volumeName := converter.GetVolumeNameByTarget(domain, disk.Disk)
+			var reasonError string
+			switch disk.Error {
+			case libvirt.DOMAIN_DISK_ERROR_NONE:
+				continue
+			case libvirt.DOMAIN_DISK_ERROR_UNSPEC:
+				reasonError = fmt.Sprintf("VM Paused due to IO error at the volume: %s", volumeName)
+			case libvirt.DOMAIN_DISK_ERROR_NO_SPACE:
+				reasonError = fmt.Sprintf("VM Paused due to not enough space on volume: %s", volumeName)
```

Comment:
I hope we won't get both at the same time :)

I think this is approach is fine as long as we will always set ErrorPolicy = "stop" on all disks, otherwise, this will break if some of the disks will be set to ignore, or even read ignore. 

What do you think Alice, is it a valid concern?

Reply:
> I hope we won't get both at the same time :)
>
I think we can theoretically get multiple IO errors for different devices. In this case, this should generate multiple k8s events for each device as it is looping for all the disks that have an error. I'll do a quick test :). If you prefer I can create a functional test like the test I did in the description.

> I think this is approach is fine as long as we will always set ErrorPolicy = "stop" on all disks, otherwise, this will break if some of the disks will be set to ignore, or even read ignore.
> 
> What do you think Alice, is it a valid concern?

So if I haven't forgotten any disk types, all the disks should get the same error policy. So they should behave the same way

Reply:
I checked what happend if we passed 2 faulty devices. Actually, it is  libvirt that reports only the first:
```ỳaml
      domain:
        devices:
          disks:
          - disk:
              bus: virtio
            name: containerdisk
          - disk:
              bus: virtio
            name: ioerror-disk
          - disk:
              bus: virtio
            name: ioerror-disk-1
   
      volumes:
      - name: ioerror-disk
        persistentVolumeClaim:
          claimName: ioerror-pvc
      - name: ioerror-disk-1
        persistentVolumeClaim:
          claimName: ioerror-pvc-1

```
Results:
```bash
$ cluster-up/kubectl.sh exec -ti virt-launcher-vm-block-jhmd7 -c compute --  virsh domblkerror default_vm-block
vdb: unspecified error
$ cluster-up/kubectl.sh  get events
3m14s       Warning   IOerror                   virtualmachineinstance/vm-block       VM Paused due to IO error at the volume: ioerror-disk


```